airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -62,10 +62,32 @@ def path(self, **kwargs) -> str:
         return f"repos/{self.repository}/{self.name}"
 
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        response_data = response.json()
-        if response_data and len(response_data) == self.page_size:
-            self._page += 1
-            return {"page": self._page}
+        link = response.headers.get("Link")
+        if link and 'rel="next"' in link:
+            response_data = response.json()
+            if response_data and len(response_data) == self.page_size:
+                self._page += 1
+                return {"page": self._page}
+
+    def should_retry(self, response: requests.Response) -> bool:
+        # We don't call `super()` here because we have custom error handling and GitHub API sometimes returns strange
+        # errors. So in `read_records()` we have custom error handling which don't require to call `super()` here.
+        return response.headers.get("X-RateLimit-Remaining") == "0" or response.status_code in (
+            500,
+            502,
+        )
+
+    def backoff_time(self, response: requests.Response) -> Optional[Union[int, float]]:
+        # This method is called if we run into the rate limit. GitHub limits requests to 5000 per hour and provides
+        # `X-RateLimit-Reset` header which contains time when this hour will be finished and limits will be reset so
+        # we again could have 5000 per another hour.
+
+        if response.status_code == 502:
+            return 0.5
+
+        reset_time = int(response.headers.get("X-RateLimit-Reset", time.time() + 60))
+        time_now = int(time.time())
+        return reset_time - time_now
```

Comment:
don't need a cast to int, you already return float
```suggestion
        reset_time = response.headers.get("X-RateLimit-Reset")
        if reset_time:
            return float(reset_time) - time.time()
        return 60
```
this looks cleaner? WDYT?

Reply:
Done.